From b136b3e2e34f9efb410a4041fa1c6f780217e0bf Mon Sep 17 00:00:00 2001
From: Wim Taymans <wtaymans@redhat.com>
Date: Wed, 28 Apr 2021 17:04:36 +0200
Subject: [PATCH 10/10] client-node: use the right port for io

To set the io on the mixer ports, we need to use the same id that
was used to add the port, not the id we use to identify the mixer
structure.

Fixes #759
---
 src/modules/module-client-node/remote-node.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/modules/module-client-node/remote-node.c b/src/modules/module-client-node/remote-node.c
index ac9fae12..61919aab 100644
--- a/src/modules/module-client-node/remote-node.c
+++ b/src/modules/module-client-node/remote-node.c
@@ -793,7 +793,7 @@ client_node_port_set_io(void *object,
 	}
 
 	if ((res = spa_node_port_set_io(mix->port->mix,
-			     direction, mix_id, id, ptr, size)) < 0) {
+			     direction, mix->mix.port.port_id, id, ptr, size)) < 0) {
 		if (res == -ENOTSUP)
 			res = 0;
 		else
-- 
2.30.2

